### PR TITLE
Add function for generating OVS external-ids

### PR DESCRIPTION
**Pull request by przemeklal**
Add `generate_external_ids()` that can be used by charms to generate
external-ids that can be set for OVS bridges and ports.
This way charms can mark those ports and bridges as managed and determine
what actions were applied by them.
This potentially enables various idempotency related functionalities.

Signed-off-by: Przemysław Lal <przemyslaw.lal@canonical.com>

### CODE REVIEW

**Comment by przemeklal** [comment]
At this stage this code is intended to be shared between openstack/charm-neutron-gateway and openstack/charm-neutron-openvswitch to avoid unnecessary duplication.
This was suggested during the review of this [change](https://review.opendev.org/c/openstack/charm-neutron-openvswitch/+/778458).

File: charmhelpers/contrib/network/ovs/__init__.py [hunk 1]
```diff
@@ -666,3 +666,28 @@ def patch_ports_on_bridge(bridge):
         # reference to PEP479 just doing a return will provide a emtpy iterator
         # and not None.
         return
+
+
+def generate_external_ids(external_id_value=None):
+    """Generate external-ids dictionary that can be used to mark OVS bridges
+    and ports as managed by the charm.
+
+    :param bridge: Value of the external-ids entry.
```
Comment (coreycb) [inline charmhelpers/contrib/network/ovs/__init__.py:675]: Should this be ":param external_id_value"?

File: charmhelpers/contrib/network/ovs/__init__.py [hunk 2]
```diff
@@ -666,3 +666,28 @@ def patch_ports_on_bridge(bridge):
...
+                   Note: 'managed' will be used if not specified.
+    :type bridge: Optional[str]
```
Comment (coreycb) [inline charmhelpers/contrib/network/ovs/__init__.py:677]: Same

Comment (przemeklal) [reply charmhelpers/contrib/network/ovs/__init__.py]: fixed

**Pull request merged**